pkg/apis/apis/v1alpha1/types_apiresourceschema.go

```diff
@@ -163,3 +163,103 @@ func (v *APIResourceVersion) SetSchema(schema *apiextensionsv1.JSONSchemaProps)
 	v.Schema.Raw = raw
 	return nil
 }
+
+const (
+	// VersionPreservationAnnotationKeySuffix is the suffix for the annotation key used to preserve fields from an API
+	// version that would otherwise be lost during round-tripping to a different API version. An example key and value
+	// might look like this: v2.conversion.apis.kcp.io/preserve: {"spec.someNewField": "someValue"}.
+	VersionPreservationAnnotationKeySuffix = ".conversion.apis.kcp.io/preserve"
+)
+
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// APIConversion contains rules to convert between different API versions in an APIResourceSchema. The name must match
+// the name of the APIResourceSchema for the conversions to take effect.
+type APIConversion struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec holds the desired state.
+	Spec APIConversionSpec `json:"spec"`
+}
+
+// APIConversionSpec contains rules to convert between different API versions in an APIResourceSchema.
+type APIConversionSpec struct {
+	// +required
+	// +listType=map
```

Comment:
For types where these attributes were *also* added to struct tags, having them on separate lines caused an error:

```
2023/01/11 07:57:40 OpenAPI code generation error: Failed executing generator: some packages had errors:
Tags in comment and struct should match for member (SubResources) of (github.com/kcp-dev/kcp/pkg/apis/apiresource/v1alpha1.CommonAPIResourceSpec)
```

Why is it correct to do this here?

Reply:
Not sure?

Reply:
Should make sure we don't need the struct-tags?

Reply:
IIRC These are required if we're using client-go like clients. The struct tags are used in https://github.com/kubernetes/kubernetes/blob/6ce055d62d8608842db18901c948c6982c74a164/staging/src/k8s.io/apimachinery/third_party/forked/golang/json/fields.go which is then used in the strategicpatch package https://github.com/kubernetes/kubernetes/blob/6ce055d62d8608842db18901c948c6982c74a164/staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/meta.go#L54

Reply:
CRDs don't support strategic merge, if that's what we're talking about? Also, none of our other code does this today. Suggest deferring to a wholesale repo follow-up if needed.